### PR TITLE
Auth store logs into remote services

### PR DESCRIPTION
**Pull request by p-jackson**
#### Changes proposed in this Pull Request

Logging in at `/log-in` logs into a number of services (depending on your account) e.g. `jetpack.com`, `gravatar.com`, etc.

This PR ensures that logging in with the auth store also logs into these services.

Code was copied from `client/state/login.js`, particularly `makeRemoteLoginRequest()`

#### Testing instructions

* `/gutenboarding`
* Open network tab and console
* `wp.auth.submitUsernameOrPassword('validuser')`
* `wp.auth.submitPassword('correctpassword')`
* Look in network tab for request to `https://jetpack.com/remote-login.php` and see it responds with lots of precious precious cookies 🍪 

You should notice that the `remote-login.php` request is a iframe/subdocument request.

### CODE REVIEW

File: packages/data-stores/src/auth/controls.ts [hunk 1]
```diff
@@ -68,5 +116,12 @@ export function createControls( clientCreds: WpcomClientCredentials ) {
 				},
 			} );
 		},
+		REMOTE_LOGIN_USER: ( { loginLinks }: RemoteLoginUserAction ) =>
+			Promise.all(
```
Comment (roo2) [inline packages/data-stores/src/auth/controls.ts]: I think that this should be awaited here!

Comment (p-jackson) [reply packages/data-stores/src/auth/controls.ts]: The `REMOTE_LOGIN_USER` function returns the promise (arrow function has no braces) so it does end up being awaited.

Comment (roo2) [reply packages/data-stores/src/auth/controls.ts]: Ahh my bad, brain fart :joy:

Comment (p-jackson) [reply packages/data-stores/src/auth/controls.ts]: ezgif-6-75adb3e22b40

**Pull request merged**